vm/parser.ts

```diff
@@ -759,8 +764,17 @@ module J2ME {
       var s = this;
       var methods_count = s.readU2();
       var m = this.methods = new Array(methods_count);
+      var ids = this.methodIDs = new Array(methods_count);
       for (var i = 0; i < methods_count; i++) {
         m[i] = s.offset;
+
+        ids[i] = [
```

Comment:
> Yes, but with this change we''re going to load eagerly all the methods of a class, even if we need only a few.
> (There are classes that contain a lot of methods that are never used)

Erm, no, with this change we're going to load eagerly only a subset of data about all the methods of a class: name, signature, and flags. And we have to load this information at startup in order to minimally link virtual methods on Klass prototypes at startup.

We don't have to do it this way, of course. We could instead make MethodInfo lighter weight and create lightweight MethodInfo objects for every method in every class we load at startup, as #1225 does. Or this branch could add a *MethodInfo.scanMethods2* method that *linkKlassMethods* calls when it's time to minimally link the methods.

But then we'd be scanning methods twice on startup: once to retrieve just their indexes, and once to retrieve their name/signature/flags.

(Ideally, we wouldn't have to minimally link methods on startup at all. And we may be able to avoid it. But neither this branch nor that one solves that problem yet. And in any case it's a minor optimization compared to the significant wins we can gain by loading only the minimum information about each method on startup.)